src/Schema/Directives/WithAvgDirective.php

```diff
@@ -0,0 +1,90 @@
+<?php
+
+namespace Nuwave\Lighthouse\Schema\Directives;
+
+use GraphQL\Language\AST\FieldDefinitionNode;
+use GraphQL\Language\AST\ObjectTypeDefinitionNode;
+use GraphQL\Type\Definition\ResolveInfo;
+use Nuwave\Lighthouse\Exceptions\DefinitionException;
+use Nuwave\Lighthouse\Execution\DataLoader\RelationAggregateLoader;
+use Nuwave\Lighthouse\Execution\DataLoader\RelationAvgLoader;
+use Nuwave\Lighthouse\Schema\AST\DocumentAST;
+use Nuwave\Lighthouse\Schema\RootType;
+use Nuwave\Lighthouse\Support\Contracts\FieldManipulator;
+use Nuwave\Lighthouse\Support\Contracts\FieldMiddleware;
+
+class WithAvgDirective extends WithRelationAggregateDirective implements FieldMiddleware, FieldManipulator
+{
+    public static function definition(): string
+    {
+        return /** @lang GraphQL */ <<<'GRAPHQL'
+"""
+Eager-load the avg of an Eloquent relation if the field is queried.
+
+Note that this does not return a value for the field, the count is simply
+prefetched, assuming it is used to compute the field value. Use `@count`
+if the field should simply return the relation count.
+"""
+directive @withAvg(
+  """
+  Specify the relationship method name in the model class.
+  """
+  relation: String!
+
+  """
+  Specify the relationship column name in the model class.
```

Comment:
This description is misleading, it is not a column of the model class but rather the relation.